File: C++/Opp/Book.cpp

#include <iostream>
#include <string>
#include <vector>

class Book {

    public:
        std::string title;
        std::string author;
        int pages;

        Book(std::string x, std::string y, int z) {
            title = x;
            author = y;
            pages = z;
        }
        void showBook() {
            std::cout << " titulo " << title << " autor: " << author << " pages " << pages << std::endl;
        }    
};

class Library {
    public:
        std::vector<Book> books = {};
        std::string name;
        Library(std::x) {
            name = x;
        }

};


int main() {

    Book kevin("Harry", "Kevin",250);
    kevin.showBook();


    return 1;
}

File: C++/Opp/Book.py


class Book:

    def __init__(self, title, author, pages):
        self.title = title
        self.author = author
        self.pages = pages

    def show(self):
        print(f"title :{self.title} author {self.author} pages {self.pages}")


class Library:
    def __init__(self):
        self.books = []

    def add_book(self,book):
        self.books.append(book)

    def show_books(self):
        for book in self.books:
            print(book.title)

    def search_book_title(self,title):
        for book in self.books:
            if book.title == title:
                print("encontrado")

    def search_book_author(self,author):
        for book in self.books:
            if book.author == author:
                print("encontrado")


harry = Book("Harry","Kevin", 250)
KevinLibrary = Library()



KevinLibrary.add_book(harry)
KevinLibrary.show_books()
KevinLibrary.search_book_title("Harry")
KevinLibrary.search_book_author("Kevin")